02-express-tutorial/logger.js

```diff
@@ -0,0 +1,11 @@
+const logger = (req, res, next) =>{//express will supply req, res, and also next
+    const method = req.method
+    const url = req.url
+    const time = new Date().getFullYear();
+    console.log(method, url, time)
+    // res.send('testing')
+    next()//pass onto next function
+    //either terminate by sending response or pass it to next middleware
+}
```

Comment:
How you described is fairly correct. It would be more correct to say this way: Logger middleware takes three arguments, The first input in the function is Request object and we give them name req. The request object is automatically assgned to req variable
The Second input in the function is Response object and we give them name res. The Request object is automatically assigned to res variable 
AND 
The next variable is assigned to the next  function which can be middle ware of call back function to the api endpoint in express. 
Remember middleware sits between api endpoint and call back function . Good work work Armani.